drivers/hwinfo/hwinfo_nrf.c

```diff
@@ -8,7 +8,7 @@
 #include <zephyr/drivers/hwinfo.h>
 #include <string.h>
 #include <zephyr/sys/byteorder.h>
-#if !defined(CONFIG_SOC_SERIES_NRF54HX) && !defined(CONFIG_BOARD_QEMU_CORTEX_M0)
+#if !defined(CONFIG_BOARD_QEMU_CORTEX_M0)
 #include <helpers/nrfx_reset_reason.h>
```

Comment:
This file cannot be included for cpuppr and cpuflpr on nRF54H20 because this leads to `#error "Unsupported device"` (and excluding them just in `tests/drivers/hwinfo/api` is not sufficient).
How about handling this with something like:
```
#if defined(CONFIG_BOARD_QEMU_CORTEX_M0) || \
	(defined(CONFIG_NRF_PLATFORM_HALTIUM) && \
	 defined(CONFIG_RISCV_CORE_NORDIC_VPR))
#define RESET_CAUSE_AVAILABLE 0
#else
#include <helpers/nrfx_reset_reason.h>
#define RESET_CAUSE_AVAILABLE 1
#endif
```
and then use `#if RESET_REASON_AVAILABLE` before `z_impl_hwinfo_get_reset_cause()`?